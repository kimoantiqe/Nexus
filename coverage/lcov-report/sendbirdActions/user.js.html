<!doctype html>
<html lang="en">
<head>
    <title>Code coverage report for sendbirdActions/user.js</title>
    <meta charset="utf-8" />
    <link rel="stylesheet" href="../prettify.css" />
    <link rel="stylesheet" href="../base.css" />
    <meta name="viewport" content="width=device-width, initial-scale=1">
    <style type='text/css'>
        .coverage-summary .sorter {
            background-image: url(../sort-arrow-sprite.png);
        }
    </style>
</head>
<body>
<div class='wrapper'>
  <div class='pad1'>
    <h1>
      <a href="../index.html">All files</a> / <a href="index.html">sendbirdActions</a> user.js
    </h1>
    <div class='clearfix'>
      <div class='fl pad1y space-right2'>
        <span class="strong">72.22% </span>
        <span class="quiet">Statements</span>
        <span class='fraction'>13/18</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">33.33% </span>
        <span class="quiet">Branches</span>
        <span class='fraction'>2/6</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">85.71% </span>
        <span class="quiet">Functions</span>
        <span class='fraction'>6/7</span>
      </div>
      <div class='fl pad1y space-right2'>
        <span class="strong">72.22% </span>
        <span class="quiet">Lines</span>
        <span class='fraction'>13/18</span>
      </div>
    </div>
  </div>
  <div class='status-line medium'></div>
<pre><table class="coverage">
<tr><td class="line-count quiet">1
2
3
4
5
6
7
8
9
10
11
12
13
14
15
16
17
18
19
20
21
22
23
24
25
26
27
28
29
30
31
32
33
34
35
36
37
38
39
40
41
42
43</td><td class="line-coverage quiet"><span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-yes">1x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-yes">8x</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-no">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span>
<span class="cline-any cline-neutral">&nbsp;</span></td><td class="text"><pre class="prettyprint lang-js">import SendBird from 'sendbird';
&nbsp;
const APP_ID = '5684560F-4764-4763-A8EB-3AD372D8B9EB';
&nbsp;
//Connect Function.
export const sbConnect = (userId, nickname) =&gt; {
    return new Promise((resolve, reject) =&gt; {
        //Create new connection.
        const sb = new SendBird({ 'appId': APP_ID });
        //Initiate connection to our sendbird backend
        sb.connect(userId, (user, error) =&gt; {
            //Handles the error
            <span class="missing-if-branch" title="else path not taken" >E</span>if (error) {
                reject('SendBird Login Failed.');
            } else {
                //Update the user profile to view his name, 
                //Second option is set to null until we implement 
                //profile pics.
<span class="cstat-no" title="statement not covered" >                sb.updateCurrentUserInfo(nickname, null, <span class="fstat-no" title="function not covered" >(u</span>ser, error) =&gt; {</span>
<span class="cstat-no" title="statement not covered" >                    if (error) {</span>
<span class="cstat-no" title="statement not covered" >                        reject('Update User Failed.');</span>
                    } else {
<span class="cstat-no" title="statement not covered" >                        resolve(user);</span>
                    }
                })
            }
        })
    })
};
&nbsp;
//Disconnect Function.
export const sbDisconnect = () =&gt; {
    return new Promise((resolve, reject) =&gt; {
        const sb = SendBird.getInstance();
        <span class="missing-if-branch" title="else path not taken" >E</span>if (sb) {
            sb.disconnect(() =&gt; {
                resolve(null);
            });
        } else {
<span class="cstat-no" title="statement not covered" >            reject('Could not get instance');</span>
        }
    })
}</pre></td></tr>
</table></pre>
<div class='push'></div><!-- for sticky footer -->
</div><!-- /wrapper -->
<div class='footer quiet pad2 space-top1 center small'>
  Code coverage
  generated by <a href="https://istanbul.js.org/" target="_blank">istanbul</a> at Mon Nov 19 2018 21:32:30 GMT-0800 (Pacific Standard Time)
</div>
</div>
<script src="../prettify.js"></script>
<script>
window.onload = function () {
        if (typeof prettyPrint === 'function') {
            prettyPrint();
        }
};
</script>
<script src="../sorter.js"></script>
</body>
</html>
