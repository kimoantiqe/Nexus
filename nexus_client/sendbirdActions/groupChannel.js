import SendBird from 'sendbird';

//Function to create 1-1 GroupChannel.
export const sbCreateChannel = (userIds, channelName) => {
    return new Promise((resolve, reject) => {
        const sb = SendBird.getInstance();
        if(sb){
            sb.GroupChannel.createChannelWithUserIds(userIds, true, channelName, null, null, null, (groupChannel, error) => {
                if (error) {
                    console.log('Create Channel failed');
                    console.log(error);
                    reject('SendBird Group Channel Creation failed');
                } else {
                    console.log('Create Channel worked');
                    resolve(groupChannel);
                }
            });
        } else {
            reject('There is no SendBird instance!');
        }
    });
};

export const sbCreateGroupChannelListQuery = () => {
    const sb = SendBird.getInstance();
    if(sb)
        return sb.GroupChannel.createMyGroupChannelListQuery();
    else    
        console.log("Group Channel Fail");
}

//Function to get a list of all the GroupChannels.
export const sbGetChannels = (channelListQuery) => {
    return new Promise((resolve, reject) => {
        channelListQuery.next(function(channelList, error){
            if (error) {
                reject("Could not retrieve the channel!");
            } else {
                resolve(channelList);
            }
        });        
    });
};

//Create a leave function so that when the user deletes the chat.
