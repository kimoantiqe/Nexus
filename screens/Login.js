import React, { Component } from "react";
import Expo from 'expo' 
import { LinearGradient } from "expo";
import {
  ActivityIndicator,
  AsyncStorage,
  StatusBar,
  StyleSheet,
  View,
  Image,
  Dimensions,
  KeyboardAvoidingView,
  Alert,
  TextInput,
  TouchableOpacity,WebView
} from "react-native";
import { Text, Button, Icon } from "native-base";

import Background from '../components/Background';

const width = Dimensions.get("window").width;
const height = Dimensions.get("window").height;

const APIcall      = require("../API_calls/APIs");

var userToken3;
export {userToken3};

export default class Login extends React.Component {
  static navigationOptions = {
    header: null,
    title: "Login"
  };

  constructor(props) {
    super(props);
    this.state = {
      username: "",
      password: "",
      userid: "",
      nickname: ""
    };
    //this. = this.modal.bind(this);
  }

  handleUsername = text => {
    this.setState({ username: text });
  };

  handlePassword = text => {
    this.setState({ password: text });
  };

  render() {
    return (
      <View>
      <Background logo= {true}/>
      <View style={styles.container}>
      <TextInput
                  placeholder="Email Address"
                  placeholderTextColor="rgba(255, 255, 255, 0.2)"
                  style={styles.input}
                  onChangeText={this.handleUsername}
                />
                <TextInput
                  placeholder="password"
                  placeholderTextColor="rgba(255, 255, 255, 0.2)"
                  secureTextEntry
                  style={styles.input}
                  onChangeText={this.handlePassword}
                />

                <TouchableOpacity style={styles.buttonContainer}>
                  <Button onPress={()=>APIcall.login(this.state.username, this.state.password, this.props)} style={styles.button}>
                    <Text style={{ fontWeight: "bold" }}>LOGIN</Text>
                  </Button>
                </TouchableOpacity>

                <View style={{ flexDirection: "row" }}>
                  <View
                    style={{
                      backgroundColor: "grey",
                      height: 1,
                      flex: 1,
                      alignSelf: "center"
                    }}
                  />
                  <Text
                    style={{
                      alignSelf: "center",
                      paddingHorizontal: 5,
                      color: "grey",
                      fontSize: 18
                    }}
                  >
                    OR
                  </Text>
                  <View
                    style={{
                      backgroundColor: "grey",
                      height: 1,
                      flex: 1,
                      alignSelf: "center"
                    }}
                  />
                </View>

                <View
                  style={{
                    flexDirection: "row",
                    marginTop: height / 70,
                    justifyContent: "center"
                  }}
                >
                  <Button
                    iconLeft
                    bordered
                    onPress = {() => this.loginFb()}
                    activeOpacity={0.5}
                    style={{
                      width: (width * 38) / 100,
                      height: height / 14,
                      borderColor: "grey"
                    }}
                  >
                    <Icon
                      style={{ fontSize: 30, color: "grey" }}
                      name="logo-facebook"
                    />
                    <Text
                      uppercase={false}
                      style={{
                        fontSize: 18,
                        fontWeight: "500",
                        color: "grey",
                        paddingLeft: width / 30
                      }}
                    >
                      Facebook
                    </Text>
                  </Button>
                </View>
              </View>
          </View>
    );
  }

  loginFb = async () => {
    try {
      const {
        type,
        token,
        expires,
        permissions,
        declinedPermissions,
      } = await Expo.Facebook.logInWithReadPermissionsAsync('1220787098061912', {
        permissions: ['public_profile'],
      });
      if (type === 'success') {
        // Get the user's name using Facebook's Graph API
        const response = await fetch(`https://graph.facebook.com/me?access_token=${token}`);
        Alert.alert('Logged in!', `Hi ${(await response.json()).name}!`);
      } else {
        // type === 'cancel'
      }
    } catch ({ message }) {
      alert(`Facebook Login Error: ${message}`);
    }
  }

}

const styles = StyleSheet.create({
  container: {
    marginBottom: height * 0.05,
    paddingLeft: width * 0.05,
    paddingRight: width * 0.05
  },
  input: {
    height: height * 0.05,
    backgroundColor: "rgba(255, 255, 255, 0.2)",
    marginBottom: height * 0.03,
    color: "#FFF",
    alignItems: "center",
    paddingHorizontal: 10,
    fontSize: 20
  },
  buttonContainer: {
    flexDirection: "row",
    width: "100%",
    height: "25%",
    justifyContent: "center",
    alignItems: "center"
  },
  button: {
    width: 200,
    backgroundColor: "transparent",
    borderColor: "grey",
    borderWidth: 1,
    borderRadius: 100,
    marginTop: 20,
    justifyContent: "center"
  }
});

