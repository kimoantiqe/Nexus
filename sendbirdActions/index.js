export * from './user';
export * from './groupChannel'