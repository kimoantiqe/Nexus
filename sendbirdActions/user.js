import SendBird from 'sendbird';

const APP_ID = '5684560F-4764-4763-A8EB-3AD372D8B9EB';

//Connect Function.
export const sbConnect = (userId, nickname) => {
    return new Promise((resolve, reject) => {
        //Create new connection.
        const sb = new SendBird({ 'appId': APP_ID });
        //Initiate connection to our sendbird backend
        sb.connect(userId, (user, error) => {
            //Handles the error
            if (error) {
                reject('SendBird Login Failed.');
            } else {
                //Update the user profile to view his name, 
                //Second option is set to null until we implement 
                //profile pics.
                sb.updateCurrentUserInfo(nickname, null, (user, error) => {
                    if (error) {
                        reject('Update User Failed.');
                    } else {
                        resolve(user);
                    }
                })
            }
        })
    })
};

//Disconnect Function.
export const sbDisconnect = () => {
    return new Promise((resolve, reject) => {
        const sb = SendBird.getInstance();
        if (sb) {
            sb.disconnect(() => {
                resolve(null);
            });
        } else {
            resolve(null);
        }
    })
}