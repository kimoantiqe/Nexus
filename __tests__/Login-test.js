import React from 'react';
import Login from '../screens/Login'

import renderer from 'react-test-renderer';

import SendBird from 'sendbird'

import {sbConnect} from '../sendbirdActions'

import { shallow } from 'enzyme';

import Enzyme from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

import { Icon } from 'react-native-elements'

import toJson from 'enzyme-to-json';

import {
    AppRegistry,
    StyleSheet,
    Text,
    View,
    Image,
    TextInput,
    TouchableHighlight,
    FlatList,
    Dimensions,
    KeyboardAvoidingView
  } from 'react-native'
 
import { Button } from 'native-base'

Enzyme.configure({ adapter: new Adapter() });
jest.mock('WebView', () => 'WebView')

test('renders correctly', () => {
  const wrapper = shallow(<Login />);
  expect(toJson(wrapper)).toMatchSnapshot();
});

test('Check Button', () => {

  const wrapper = shallow(<Login />);

  wrapper.find(Button).first().props().onPress();

  expect(wrapper.find(Button).first().props().onPress()).toHaveLength(1);
});