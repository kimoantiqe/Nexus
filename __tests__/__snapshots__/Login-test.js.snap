// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders correctly 1`] = `
<Component>
  <Background
    logo={true}
  />
  <Component
    style={
      Object {
        "marginBottom": 66.7,
        "paddingLeft": 37.5,
        "paddingRight": 37.5,
      }
    }
  >
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="Email Address"
      placeholderTextColor="rgba(255, 255, 255, 0.2)"
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.2)",
          "color": "#FFF",
          "fontSize": 20,
          "height": 66.7,
          "marginBottom": 40.019999999999996,
          "paddingHorizontal": 10,
        }
      }
    />
    <TextInput
      allowFontScaling={true}
      onChangeText={[Function]}
      placeholder="password"
      placeholderTextColor="rgba(255, 255, 255, 0.2)"
      secureTextEntry={true}
      style={
        Object {
          "alignItems": "center",
          "backgroundColor": "rgba(255, 255, 255, 0.2)",
          "color": "#FFF",
          "fontSize": 20,
          "height": 66.7,
          "marginBottom": 40.019999999999996,
          "paddingHorizontal": 10,
        }
      }
    />
    <TouchableOpacity
      activeOpacity={0.2}
      style={
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "height": "25%",
          "justifyContent": "center",
          "width": "100%",
        }
      }
    >
      <Styled(Button)
        onPress={[Function]}
        style={
          Object {
            "backgroundColor": "transparent",
            "borderColor": "grey",
            "borderRadius": 100,
            "borderWidth": 1,
            "justifyContent": "center",
            "marginTop": 20,
            "width": 200,
          }
        }
      >
        <Styled(Text)
          style={
            Object {
              "fontWeight": "bold",
            }
          }
        >
          LOGIN
        </Styled(Text)>
      </Styled(Button)>
    </TouchableOpacity>
    <Component
      style={
        Object {
          "flexDirection": "row",
        }
      }
    >
      <Component
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "grey",
            "flex": 1,
            "height": 1,
          }
        }
      />
      <Styled(Text)
        style={
          Object {
            "alignSelf": "center",
            "color": "grey",
            "fontSize": 18,
            "paddingHorizontal": 5,
          }
        }
      >
        OR
      </Styled(Text)>
      <Component
        style={
          Object {
            "alignSelf": "center",
            "backgroundColor": "grey",
            "flex": 1,
            "height": 1,
          }
        }
      />
    </Component>
    <Component
      style={
        Object {
          "flexDirection": "row",
          "justifyContent": "center",
          "marginTop": 19.057142857142857,
        }
      }
    >
      <Styled(Button)
        activeOpacity={0.5}
        bordered={true}
        iconLeft={true}
        onPress={[Function]}
        style={
          Object {
            "borderColor": "grey",
            "height": 95.28571428571429,
            "width": 285,
          }
        }
      >
        <Styled(Icon)
          name="logo-facebook"
          style={
            Object {
              "color": "grey",
              "fontSize": 30,
            }
          }
        />
        <Styled(Text)
          style={
            Object {
              "color": "grey",
              "fontSize": 18,
              "fontWeight": "500",
              "paddingLeft": 25,
            }
          }
          uppercase={false}
        >
          Facebook
        </Styled(Text)>
      </Styled(Button)>
    </Component>
  </Component>
</Component>
`;
